test/quantization/test_workflow_module.py

```diff
@@ -862,22 +862,36 @@ def test_backward_per_tensor(self, device, X):
         Y_prime.backward(dout)
         np.testing.assert_allclose(dX.cpu(), X.grad.cpu().detach().numpy(), rtol=tolerance, atol=tolerance)
 
-    def _test_forward_per_tensor_cachemask_impl(self, device):
-        for torch_type in (torch.qint8, torch.quint8):
-            Xs = (torch.randn(4, 8, device=device), torch.randn(4, 16, device=device)[:, ::2])
-            # pick the scale + zp so that some values get clipped
-            for X in Xs:
-                obs = torch.quantization.MinMaxObserver(torch_type)
-                obs(X * 0.75)
-                scale, zero_point = obs.calculate_qparams()
-                scale, zero_point = float(scale), int(zero_point)
-                quant_min, quant_max = obs._calculate_qmin_qmax()
+    def test_forward_backward_per_tensor_cachemask_with_amp(self):
```

Comment:
great!

Can we add just one more numerics test, where we ensure the fp16 calculation is numerically correct?  Something like
1. audit the reference function to ensure it handles fp16 correctly
2. create a hand crafted input and verify the reference function works for it correctly
3. pass the hand crafted input into the real function and verify it matches reference

some cases to test would be
1. normal looking tensors
2. overflow
3. underflow